### ISSUE TITLE
[a11y]: SidePanel should provide "aria-label" to support the case when title is not provided

### ISSUE THREAD

**Issue by hyungskim** [carbon-design-system/ibm-products#6906]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.59.0

### React version

v17.0.2

### Automated testing tool and ruleset

IBM Accessibility Checker - Latest Deployment / IBM Accessibility v7.3

### Description

carbon/ibm-products needs to support providing an "aria-label" for SidePanel to support the case when title is not provided. Noticed title would fix the issue and SidePanel seems to use title as "aria-label" internally. Our use case below from Catalog UI is using the SidePanel without title so "aria-label" should be supported.

<img width="1639" alt="Image" src="https://github.com/user-attachments/assets/c2f0cff4-8f54-415b-a35e-79ff3583993f" />

### Reproduction/example

https://codesandbox.io/p/sandbox/new?file=%2Fsrc%2FApp.js

### Steps to reproduce

"@carbon/ibm-products": "2.59.0"

Use SidePanel without title as below

```
    <SidePanel
      className={styles.sidePanel}
      open={artifactPanelState?.open || false}
      size="sm"
      .......
      closeIconDescription={intl.formatMessage(messages.close)}
      slideIn
      selectorPageContent={selectorPageContent}
    >
      {props.children}
    </SidePanel>
```

<img width="1639" alt="Image" src="https://github.com/user-attachments/assets/96db034b-c121-47dd-a465-6b5102092f29" />

   

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

### Acceptance Criteria

- [x] Include `aria-label`
- [x] Use default `aria-label`, if it is not provided use the `title`

**Comment by wkeese** [comment]
Is this still an issue?  I already see an aria-label on https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic.
